### PR TITLE
Move to using WDA xml source method

### PR DESCRIPTION
**Pull request by imurchie**
XML output is the same as it was.

Addresses https://github.com/appium/appium/issues/7382

### CODE REVIEW

File: lib/commands/source.js [hunk 1]
```diff
@@ -15,77 +14,18 @@ commands.getPageSource = async function () {
...
+  let method = 'GET';
+  let endpoint = `/source/xml`;
   let srcTree = await this.proxyCommand(endpoint, method);
 
-  // translate the returned value into what Appium expects
-  let src = getSourceXml(getTreeForXML(srcTree));
+  // need to add <AppiumAUT /> tags, after the xml header
+  //   <?xml version="1.0" encoding="UTF-8"?>
+  let src = srcTree.tree;
+  let match = XML_HEADER_REGEXP.exec(src);
```
Comment (mykola-mokhnach) [inline lib/commands/source.js]: A kitten dies somewhere when one parses xml with regexp. Why not to use the standard DOM parser to modify the document?

File: lib/commands/source.js [hunk 2]
```diff
@@ -15,77 +14,18 @@ commands.getPageSource = async function () {
...
 
-  return src;
+  return `${match[0]}<AppiumAUT>${src.substring(match[0].length)}</AppiumAUT>`;
```
Comment (mykola-mokhnach) [inline lib/commands/source.js]: can we move the name of this special enclosing tag into a constant?

File: lib/commands/source.js (hunk 1 above)
Comment (imurchie) [inline lib/commands/source.js]: I was loathe to add full dom parsing just to stick that in. Plus I like regular expressions and I _hate_ XML.

But we probably should do it. Bleh.

**Comment by imurchie** [comment]
Alrighty. All XMLified.

**Comment by mykola-mokhnach** [comment]
its-perfect

**Comment by mykola-mokhnach** [comment]
@imurchie you may consider using **const** instead of **let** by default while defining local variables which are not gonna be changed anymore. Small thing, but makes the code a bit safer.

**Comment by imurchie** [comment]
How safer? `const` just means you can't reassign the value of a variable, which is not much safety for the price of some odd semantics. It also lulls one into thinking that the variable is "constant" in a greater sense than it is. I vote against `const` outside of true constants.

**Comment by pluie-cloud** [comment]
this api is too slowly

**Pull request merged**